### PR TITLE
Frontend view single weblink 2

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for the accidently merged and then reverted PR #325

### CODE REVIEW

**Comment by alikon** [comment]
i'm unable to select an existing weblink when i create a new menu item of the  weblink item type

**Comment by astridx** [comment]
@alikon 
Thank you very much for testing. 

For choosing a already created weblink a file in the media folder is responsible. 
I just installed a new joomla-cms and applied the patch. I realized, that the data I created in the media folder, are not created via the patch tester.

I added  this folder in the xml-file: https://github.com/joomla-extensions/weblinks/pull/336/files#diff-ca7e22c702629b3fa6a44ae18e15ee3aR36

Do i forgot something?

**Comment by alikon** [comment]
i'll double check, cause i've used com_patchtester...

**Comment by alikon** [comment]
added the js file all works fine

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 0c885e1c39bf5766e9a269176222a0afd2492097<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/336">issues.joomla.org/tracker/weblinks/336</a>.</sub>

**Comment by astridx** [comment]
@alikon Thank you very much.
Does any one know why patch tester did not add the files from the media folder automaticly.

**Comment by zero-24** [comment]
@astridx i have just tested this but it looks like the fields are missing in the frontend. I would suggest to use the onContent over the onWeblink events. For the normal backend we are also using the onContent events.

If i use the onContent events it is working great.

**Comment by zero-24** [comment]
I'm just thinking loud.

Don't we need to wait to release a version with the new strings until they are added to the CMS and forwarded to the tranlation teams?

So we can ship this things first if this works and is merged but the strings are merged to the CMS too so the TT's got a chance to translate them?

This would mean this feature (this an my PR) is part of an weblinks 3.7.1 realase (short after 3.7.1 is out) ? Or I'm missing something?

cc @yvesh ?

**Comment by astridx** [comment]
@zero-24  Thank you. I changed the events.

**Comment by zero-24** [comment]
Great. Can you also alpha order the new language strings in the language file? If i get some time when i'm back home i can send you some tweeks for the layout files (use opening and closing PHP statments per line).

If that is sorted we need to take a decision over the general issue i have raised above.

Than this is ready to be released (together with my other PR :)

:+1:

File: src/administrator/components/com_weblinks/views/weblinks/tmpl/modal.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+							</div>
+						</td>
+						<td class="small hidden-phone">
+							<?php echo $this->escape($item->access_level); ?>
+						</td>
+						<td class="small">
+							<?php echo JLayoutHelper::render('joomla.weblinks.language', $item); ?>
+						</td>
+						<td class="nowrap small hidden-phone">
+							<?php // todo echo JHtml::_('date', $item->created, JText::_('DATE_FORMAT_LC4')); ?>
```
Comment (zero-24) [inline src/administrator/components/com_weblinks/views/weblinks/tmpl/modal.php]: what is the reason for the todo did this not work?

File: src/components/com_weblinks/views/weblink/tmpl/default.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		</h2>
+	</div>
+
+	<?php // Content is generated by content plugin event "onContentAfterTitle" ?>
+	<?php echo $this->item->event->afterDisplayTitle; ?>
+
+	<?php // Content is generated by content plugin event "onContentBeforeDisplay" ?>
+	<?php echo $this->item->event->beforeDisplayContent; ?>
+
+	<div itemprop="articleBody">
```
Comment (zero-24) [inline src/components/com_weblinks/views/weblink/tmpl/default.php]: I'm not sure here about `articleBody`

**Comment by zero-24** [comment]
@astridx i have just send you a PR: https://github.com/astridx/weblinks/pull/2 wich should fix the most smal errors.

**Comment by zero-24** [comment]
Please take a look into the two inline comments I made here.

Based on the Feedback by @yvesh in the Weblinks Chat I'm going to change the milestone to 3.7.1 so this is going to be merged short after 3.7.0 weblinks is released and the language strings are getting moved as quick as possible to the CMS to be included in 3.7.1 too. Thanks!

**Comment by astridx** [comment]
If I am right, there is no open issue or question in the conversation of this PR at the moment, right?

**Comment by zero-24** [comment]
The only one is this line +	<div itemprop="articleBody"> i'm not sure as it is no article. Bu i'm no expert on this.

File: src/components/com_weblinks/views/weblink/tmpl/default.php [hunk 3]
```diff
@@ -23,7 +23,7 @@
 	<?php echo $this->item->event->afterDisplayTitle; ?>
 	<?php // Content is generated by content plugin event "onContentBeforeDisplay" ?>
 	<?php echo $this->item->event->beforeDisplayContent; ?>
-	<div itemprop="articleBody">
+	<div itemprop="weblinkBody">
```
Comment (astridx) [inline src/components/com_weblinks/views/weblink/tmpl/default.php]: I changed the itemprop to "weblinkBody"

Comment (zero-24) [inline src/components/com_weblinks/views/weblink/tmpl/default.php]: I'm not sure here. As i don't know how that itemprop thing is working.

File: src/components/com_weblinks/views/weblink/tmpl/default.php [hunk 4]
```diff
@@ -23,7 +23,7 @@
 	<?php echo $this->item->event->afterDisplayTitle; ?>
 	<?php // Content is generated by content plugin event "onContentBeforeDisplay" ?>
 	<?php echo $this->item->event->beforeDisplayContent; ?>
-	<div itemprop="weblinkBody">
+	<div itemprop="articleBody">
```
Comment (astridx) [inline src/components/com_weblinks/views/weblink/tmpl/default.php]: I rechanged the itemprop again. 
If I am right, this is only for search engins. Search engins can use this for give a sematnic to the html-tag. 
You can use your one vocabulary but search engines favour the schema.org vocabulary.
If I search the website http://schema.org/docs/gs.html#microdata_why I find no name that suite more than articlebody: http://schema.org/Article

Comment (zero-24) [inline src/components/com_weblinks/views/weblink/tmpl/default.php]: Than i think we are good to go with that until someone comes with something better. :+1: Thanks for checking!

File: src/administrator/components/com_weblinks/models/weblinks.php [hunk 5]
```diff
@@ -41,6 +41,7 @@ public function __construct($config = array())
 				'ag.title', 'access_level',
 				'created', 'a.created',
 				'created_by', 'a.created_by',
+				'created', 'a.created',
```
Comment (infograf768) [inline src/administrator/components/com_weblinks/models/weblinks.php]: already exists 2 lines above

File: src/administrator/language/en-GB/en-GB.com_weblinks.sys.ini [hunk 6]
```diff
@@ -13,13 +13,18 @@ COM_WEBLINKS_CATEGORY_EDIT_TITLE="Category Manager: Edit A Web Links Category"
 COM_WEBLINKS_CATEGORY_VIEW_DEFAULT_DESC="Displays a list of Web Links for a category."
 COM_WEBLINKS_CATEGORY_VIEW_DEFAULT_OPTION="Default"
 COM_WEBLINKS_CATEGORY_VIEW_DEFAULT_TITLE="List Web Links in a Category"
+COM_WEBLINKS_CHANGE_WEBLINK="Select or Change weblink"
```
Comment (infograf768) [inline src/administrator/language/en-GB/en-GB.com_weblinks.sys.ini]: weblink should be capitalised = Select or Change Weblink

File: src/media/js/admin-weblinks-modal.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
+/**
+ * @copyright  Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
```
Comment (infograf768) [inline src/media/js/admin-weblinks-modal.js]: All copyright dates should be 2005 - 2017 , not 2016

File: src/administrator/language/en-GB/en-GB.com_weblinks.sys.ini (hunk 6 above)
Comment (yvesh) [inline src/administrator/language/en-GB/en-GB.com_weblinks.sys.ini]: Actually it should be Web Link

File: src/administrator/language/en-GB/en-GB.com_weblinks.sys.ini [hunk 8]
```diff
@@ -13,13 +13,18 @@ COM_WEBLINKS_CATEGORY_EDIT_TITLE="Category Manager: Edit A Web Links Category"
...
 COM_WEBLINKS_CONTENT_TYPE_WEBLINK="Web Link"
 COM_WEBLINKS_CONTENT_TYPE_CATEGORY="Web Links Category"
+COM_WEBLINKS_FIELD_SELECT_WEBLINK_DESC="Select the desired weblink from the list."
```
Comment (yvesh) [inline src/administrator/language/en-GB/en-GB.com_weblinks.sys.ini]: Please change it everywhere Web Link

**Comment by astridx** [comment]
@infograf768 Thank you for reviewing.
I fixed all issues.

**Comment by infograf768** [comment]
I posted for you some files to implement associations when editing the weblink and in its manager on https://github.com/joomla-extensions/weblinks/pull/346

**Comment by infograf768** [comment]
It is now a full .diff which would make this PR as well as #346 obsolete.

See https://github.com/joomla-extensions/weblinks/pull/346#issuecomment-301323940

**Comment by astridx** [comment]
close because of https://github.com/joomla-extensions/weblinks/pull/349

**Pull request closed without merging**